app/src/main/java/org/anitab/mentorship/remote/datamanager/AuthDataManager.kt

```diff
@@ -34,6 +34,10 @@ class AuthDataManager(private val dispatcher: CoroutineDispatcher = Dispatchers.
      * @return an Observable CustomResponse
      */
     suspend fun register(register: Register): CustomResponse {
-        return withContext(dispatcher) { apiManager.authService.register(register) }
+        try {
```

Comment:
Revert and make changes in `CommonUtils.getErrorResponse` because according to the log it's a GSON issue.

Reply:
> 

the error message is "com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $"

it seems like JSON string wich is returned with error message has not the correct structure to be parsed into an instance of CustomResponse

Gson is expecting JSON string to begin with an object opening brace. e.g.

{
But the string you have passed to it starts with an open quotes
"

This means that my PR is necessary to avoid this bug due to a server error.

Reply:
Yes but you are throwing a generic exception so there is no way to know why the error actually occurred, for instance even if it is an `IOExpection` the snackbar in this case will say `something went wrong`. However before the change it would say `Please check your internet connection` .

Reply:
Also the GSON instance you are talking about is in `CommonUtils.getErrorResponse`